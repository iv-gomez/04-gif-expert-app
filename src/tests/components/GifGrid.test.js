import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import "@testing-library/jest-dom";

jest.mock("../../hooks/useFetchGifs");

describe("<GifGrid> group", () => {
  const category = "one punch";
  test("should render ok", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("it should show items when the image are loaded with the useFetchGifs", () => {
    const gifs = [
      {
        id: "abc",
        url: "https://localhost/some.jpg",
        title: "image title",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
