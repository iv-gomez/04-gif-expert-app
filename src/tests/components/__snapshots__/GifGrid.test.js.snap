// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid> group it should show items when the image are loaded with the useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    one punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="image title"
      url="https://localhost/some.jpg"
    />
  </div>
</Fragment>
`;

exports[`<GifGrid> group should render ok 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    one punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
